scapy/layers/tls/extensions.py

```diff
@@ -780,7 +780,7 @@ def i2len(self, pkt, i):
 
     def getfield(self, pkt, s):
         tmp_len = self.length_from(pkt)
-        if tmp_len is None:
+        if tmp_len <= 0:
```

Comment:
Are you certain that `.length_from()` will never return `None`?

Reply:
When used during dissection (it's in `getfield` here), no because the `extlen` field replaces None with 0.
The bug was caused because `lenght_from` would typically return `-2` which wasn't catched here.

Reply:
Just to be safe I've added a `or 0`.